runtime/Python3/src/antlr4/TokenStreamRewriter.py

```diff
@@ -165,28 +172,31 @@ def _reduceToSingleOperationPerIndex(self, rewrites):
         for i, iop in enumerate(rewrites):
             if any((iop is None, not isinstance(iop, TokenStreamRewriter.InsertBeforeOp))):
                 continue
-            prevInserts = [op for op in rewrites[:i] if isinstance(iop, TokenStreamRewriter.InsertBeforeOp)]
-            for prevIop in prevInserts:
-                if prevIop.index == iop.index:
+            prevInserts = [op for op in rewrites[:i] if isinstance(op, TokenStreamRewriter.InsertBeforeOp)]
+            for i, prevIop in enumerate(prevInserts):
+                if prevIop.index == iop.index and type(prevIop) is TokenStreamRewriter.InsertBeforeOp:
                     iop.text += prevIop.text
                     rewrites[i] = None
+                elif prevIop.index == iop.index and type(prevIop) is TokenStreamRewriter.InsertAfterOp:
+                    iop.text = prevIop.text + iop.text
+                    rewrites[i] = None
             # look for replaces where iop.index is in range; error
             prevReplaces = [op for op in rewrites[:i] if isinstance(op, TokenStreamRewriter.ReplaceOp)]
             for rop in prevReplaces:
                 if iop.index == rop.index:
                     rop.text = iop.text + rop.text
                     rewrites[i] = None
                     continue
-                if all((iop.index >= rop.index, iop.index <= rop.index)):
+                if all((iop.index >= rop.index, iop.index <= rop.last_index)):
                     raise ValueError("insert op {} within boundaries of previous {}".format(iop, rop))
 
-            reduced = {}
-            for i, op in enumerate(rewrites):
-                if op is None: continue
-                if reduced.get(op.index): raise ValueError('should be only one op per index')
-                reduced[op.index] = op
+        reduced = {}
+        for i, op in enumerate(rewrites):
+            if op is None: continue
+            if reduced.get(op.index): raise ValueError('should be only one op per index')
+            reduced[op.index] = op
 
-            return reduced
+        return reduced
```

Comment:
Just to find a mis-indented block ...... I don't know what to say

Reply:
@youkaichao welcome to open source, I did not write this code, which strictly speaking is not part of the runtime. I guess @vlad needed it in Python 2 and lazily copied it to Python 3 without testing it. Thanks for your help!